app/code/BVMyBlog/Blog/Model/BlockRepository.php

```diff
@@ -0,0 +1,216 @@
+<?php
+declare(strict_types=1);
+
+namespace BVMyBlog\Blog\Model;
+
+use BVMyBlog\Blog\Api\BlockRepositoryInterface;
+use BVMyBlog\Blog\Model\ResourceModel\Block as ResourceBlock;
+use BVMyBlog\Blog\Model\ResourceModel\Block\CollectionFactory as BlockCollectionFactory;
+use Magento\Framework\Api\DataObjectHelper;
+use Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface;
+use Magento\Framework\Exception\CouldNotDeleteException;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Reflection\DataObjectProcessor;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Framework\Api\SearchResultsInterface;
+use Magento\Framework\Api\SearchResultsInterfaceFactory;
+use BVMyBlog\Blog\Api\Data\BlockInterfaceFactory;
+use BVMyBlog\Blog\Api\Data\BlockInterface;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use BVMyBlog\Blog\Model\ResourceModel\Block\Collection;
+use Magento\Framework\App\ObjectManager;
+use BVMyBlog\Blog\Model\Api\SearchCriteria\BlockCollectionProcessor;
+
+/**
+ * Model Block Repository
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class BlockRepository implements BlockRepositoryInterface
```

Comment:
All fields should be **private** by default. Refer to [2.7 Class design](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html#2-class-design)